js/map.js

```diff
@@ -50,20 +74,21 @@ function initialize() {
 
     var nodeVisible;
     var nodeData = new Array();
-
+    var nodeDataBySSID = new Array();
     //loop through each node
     for (var key in data.nodeList) {
       var results = data.nodeList[key];
 
       nodeVisible = 1; //Default all nodes to visible
-      nodeData[results['name']] = results;
-
       //Adjust visibility based on value and option variable
       if (!results['status']) results['status'] = 'active';
-      if (results['status'] == 'active' && !filterActive) nodeVisible = 0;
-      if (results['status'] == 'proposed' && !filterProposed) nodeVisible = 0;
+      if (results['status'] == 'active' && !filterActive) results['status']='invisible';
+      if (results['status'] == 'proposed' && !filterProposed) results['status']='invisible';
+
+      nodeData[results['name']] = results;
+      if (results['mode'] != 'station' && results['ssid']!=undefined) nodeDataBySSID[results['ssid']]=results;
```

Comment:
lines like this one are somewhat hard to read. I'd welcome a comment explaining what the expected states are and the logic for the assignment:

ie. `// mode is either 'station' or 'node'. Add nodes to the map for lookup by SSID`